Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -83,79 +162,258 @@ public override void Prepare()
             }
             catch (Exception e)
             {
-                Console.WriteLine("No such dir or could not delete dir with exception {0}.\nIf dir still exists, this analyzer will append data to existing DB.", e);
+                Console.WriteLine("Directory entered does not exist, creating directory for DB.", e);
             }
 
             var accessor = KeyValueStoreAccessor.Open(storeDirectory: OutputDirPath);
 
             if (accessor.Succeeded)
             {
-                Accessor = accessor.Result;
+                m_accessor = accessor.Result;
                 m_accessorSucceeded = true;
             }
             else
             {
                 Console.Error.WriteLine("Could not access RocksDB datastore. Exiting analyzer.");
             }
+            m_eventCount = 0;
         }
 
         /// <inheritdoc/>
         public override int Analyze()
         {
-            if (!m_accessorSucceeded)
-            {
-                return 0;
-            }
-
-            Analysis.IgnoreResult(
-              Accessor.Use(database =>
-              {
-                  foreach (var invEvent in m_invocationEventList.BXLInvEventList)
-                  {
-                      var eq = new EventTypeQuery
-                      {
-                          EventTypeID = (int)ExecutionEventId.DominoInvocation,
-                          UUID = invEvent.UUID
-                      };
-
-                      database.Put(eq.ToByteArray(), invEvent.ToByteArray());
-                  }
-              })
-            );
-
-            Accessor.Dispose();
+            m_accessor.Dispose();
+            Console.WriteLine("Num events ingested = {0}", m_eventCount);
+            Console.WriteLine("Total time elapsed: {0} seconds", m_stopWatch.ElapsedMilliseconds / 1000.0);
             return 0;
         }
 
+        /// <inheritdoc/>
+        public override void Dispose()
+        {
+            var ec = new EventCount_XLGpp
+            {
+                Value = (uint)m_eventCount
+            };
+            WriteToDb(Encoding.ASCII.GetBytes("EventCount"), ec.ToByteArray());
+        }
+
         /// <inheritdoc/>
         public override bool CanHandleEvent(ExecutionEventId eventId, uint workerId, long timestamp, int eventPayloadSize)
         {
-            if (eventId.Equals(ExecutionEventId.DominoInvocation))
+            if (m_accessorSucceeded && eventId != ExecutionEventId.ObservedInputs)
             {
-                WorkerID = workerId; // store workerID to pass into protobuf object to identify this event
                 return true;
             }
-            return false; // return false to keep the event from being parsed
+            return false;

```

Comment:
nit: `return <condition>`

Reply:
Ahh, I think this used to hold more logic, but not anymore. 

---
In reply to: [308907026](https://github.com/microsoft/BuildXL/pull/671#discussion_r308907026) [](ancestors = 308907026)